tests/client/wire_protocol.rs

```diff
@@ -22,7 +22,7 @@ fn insert_single_key_doc() {
 
             let cm = match res {
                 Ok(message) => message,
-                Err(_) => panic!("Could not create message!"),
+                _ => panic!("Could not create message!"),
```

Comment:
I find it interestint that clippy prefers this; to me, explicitly putting `Err` is much more clear. More generally, I feel like this would help catch potential logic errors if an enum has a new variant added to it (this of course isn't going to happen with Result, but I'm assuming this lint applies to all enums and not just Result.

Is this lint something you feel strongly about? My personal preference would be not to include this one.